### ISSUE TITLE
Line wrapping should be more clever about whitespace

### ISSUE THREAD

**Issue by darklajid** [maebert/jrnl#222]
So, stumbled upon this project on HN, just installed it.

I could easily fall in love. Ignoring a random limitation that bugs me (not jrnl's fault, but .. a ' is a) common in english text and b) not usable without quotes on the command line) I think this is ugly and could be improved:

````
$ jrnl 
[Nothing saved to file]
2014-06-28 19:34 So I really like this idea.
| I've lost track of what I did and wanted to do quite some time ago, let's see
|  if this tool can be used to bring order to chaos. I'm not overly convinced, 
| but if this doesn't work, the _tool_ is not to blame I guess....
````

Note that the line wrap ends up starting a line with a whitespace. I like the wrap at 73 (or whatever) characters, but spaces probably should be swallowed in that case. Is this intentional?

**Comment by philipsd6** [comment]
I don't think this is invalid at all -- it really makes the output visually "clunky" and it's easily fixed by PR #376 without breaking entries with indented code (or poems for that matter.)

**Comment by MinchinWeb** [comment]
Turning on `drop_whitespace=True` on the textwrapper has the side effect of creating blank lines, which are not "indented". This results in a break in the side bar. Eg.

```
2016-01-11 21:23 Test #222
| I've lost track of what I did and wanted to do quite some time ago, let's see
| if this tool can be used to bring order to chaos. I'm not overly convinced,
| but if this doesn't work, the _tool_ is not to blame I guess....

| +test
```

060565c fixes this by replacing blank lines with the sidebar. So the output becomes:

```
2016-01-11 21:23 Test #222
| I've lost track of what I did and wanted to do quite some time ago, let's see
| if this tool can be used to bring order to chaos. I'm not overly convinced,
| but if this doesn't work, the _tool_ is not to blame I guess....
| 
| +test
```

**Comment by philipsd6** [comment]
@MinchinWeb I'm not sure what's going on here -- my pull request was specifically created to handle the sidebar issue you noted above.

Using the code currently in master:
![image](https://cloud.githubusercontent.com/assets/908201/12431792/62bf9172-bec5-11e5-9c3f-5ee5a6fd2f67.png)

**Comment by maebert** [comment]
Merged in #420!
